File: public/examples/go/basic/04_control_flow.go
package main

import "fmt"

func main() {
    score := 85

    // If-Else
    if score >= 90 {
        fmt.Println("Grade: A")
    } else if score >= 80 {
        fmt.Println("Grade: B")
    } else {
        fmt.Println("Grade: C")
    }

    // Switch
    switch {
    case score > 80:
        fmt.Println("Well done!")
    default:
        fmt.Println("Keep trying.")
    }

    // For loop (the only loop in Go)
    fmt.Println("Counting:")
    for i := 0; i < 3; i++ {
        fmt.Println(i)
    }
}

File: public/examples/cpp/basic/04_control_flow.cpp
#include <iostream>

int main() {
    int score = 85;

    // If-Else
    if (score >= 90) {
        std::cout << "Grade: A" << std::endl;
    } else if (score >= 80) {
        std::cout << "Grade: B" << std::endl;
    } else {
        std::cout << "Grade: C" << std::endl;
    }

    // Switch
    char grade = 'B';
    switch(grade) {
        case 'A': std::cout << "Excellent!" << std::endl; break;
        case 'B': std::cout << "Well done" << std::endl; break;
        default: std::cout << "Keep trying" << std::endl;
    }

    // Loops
    for (int i = 0; i < 3; ++i) {
        std::cout << "Count: " << i << std::endl;
    }

    return 0;
}
